private/bufpkg/bufcheck/client.go

```diff
@@ -389,7 +389,6 @@ func ignoreAnnotation(
 			return true, nil
 		}
 	}
-	// TODO: Is this right? Does this properly encapsulate old extraIgnoreDescriptors logic?
```

Comment:
I did an audit of this by going through our old lint and breaking handlers and the locations that were set. All the lint extra locations are captured by protosourcepath checks. Breaking checks have previous locations and/or file names (depending on what’s applicable) set on them, and this logic ensures these are checked for ignores.